File: pdu/python/hako_srv_msgs/pdu_cdr_conv_AckEventResponsePacket.py
from .pdu_pytype_AckEventResponsePacket import AckEventResponsePacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_AckEventResponse import *
from ..hako_srv_msgs.pdu_cdr_conv_ServiceResponseHeader import *



def py_to_cdr_body_AckEventResponsePacket(writer: CdrWriter, src: AckEventResponsePacket):
    py_to_cdr_body_ServiceResponseHeader(writer, src.header)
    py_to_cdr_body_AckEventResponse(writer, src.body)


def cdr_body_to_py_AckEventResponsePacket(reader: CdrReader, dst: AckEventResponsePacket):
    cdr_body_to_py_ServiceResponseHeader(reader, dst.header)
    cdr_body_to_py_AckEventResponse(reader, dst.body)
    return dst


def py_to_cdr_AckEventResponsePacket(src: AckEventResponsePacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_AckEventResponsePacket(writer, src)
    return writer.bytes()


def cdr_to_py_AckEventResponsePacket(cdr_payload) -> AckEventResponsePacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = AckEventResponsePacket()
    return cdr_body_to_py_AckEventResponsePacket(reader, dst)

File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_AckEventResponsePacket.js
import { AckEventResponsePacket } from './pdu_jstype_AckEventResponsePacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { AckEventResponse } from '../hako_srv_msgs/pdu_jstype_AckEventResponse.js';
import { PduAckEventResponseConverter } from '../hako_srv_msgs/pdu_cdr_conv_AckEventResponse.js';
import { ServiceResponseHeader } from '../hako_srv_msgs/pdu_jstype_ServiceResponseHeader.js';
import { PduServiceResponseHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.js';


export class PduAckEventResponsePacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { AckEventResponsePacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceResponseHeaderConverter.to_cdr_body(writer, src.header);
        PduAckEventResponseConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { AckEventResponsePacket } dst
     * @returns { AckEventResponsePacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceResponseHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduAckEventResponseConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { AckEventResponsePacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { AckEventResponsePacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new AckEventResponsePacket());
    }
}
